src/ansible_compat/runtime.py

```diff
@@ -284,23 +284,38 @@ def install_requirements(self, requirement: str, retry: bool = False) -> None:
                 raise AnsibleCommandError(run)
 
     def prepare_environment(
-        self, required_collections: Optional[Dict[str, str]] = None, retry: bool = False
+        self,
+        required_collections: Optional[Dict[str, str]] = None,
+        retry: bool = False,
+        install_local: bool = False,
     ) -> None:
         """Make dependencies available if needed."""
         if required_collections is None:
             required_collections = {}
 
         self.install_requirements("requirements.yml", retry=retry)
 
+        destination = f"{self.cache_dir}/collections" if self.cache_dir else None
         for name, min_version in required_collections.items():
             self.install_collection(
                 f"{name}:>={min_version}",
-                destination=f"{self.cache_dir}/collections" if self.cache_dir else None,
+                destination=destination,
             )
 
         self._prepare_ansible_paths()
-        # install role if current project looks like a standalone role
-        self._install_galaxy_role(self.project_dir, ignore_errors=True)
+
+        if not install_local:
+            return
+
+        if os.path.exists("galaxy.yml"):
+            # molecule scenario within a collection
+            self.install_collection_from_disk(".", destination=destination)
+        elif os.path.exists("../../galaxy.yml"):
+            # molecule scenario located within a role part of a collection
```

Comment:
This is not exactly true since the condition will trigger in all cases where the `galaxy.yml` is two levels up from the CWD and not only for roles.